lib/unleash/feature_toggle.rb

```diff
@@ -51,33 +55,42 @@ def self.disabled_variant
 
     private
 
-    def resolve_variant(context, toggle_enabled)
+    def resolve_variant(context, toggle_enabled, variants, group_id)
       return Unleash::FeatureToggle.disabled_variant unless toggle_enabled
-      return Unleash::FeatureToggle.disabled_variant if sum_variant_defs_weights <= 0
+      return Unleash::FeatureToggle.disabled_variant if sum_variant_defs_weights(variants) <= 0
 
-      variant_from_override_match(context) || variant_from_weights(context, resolve_stickiness)
+      variant_from_override_match(context, variants) || variant_from_weights(context, resolve_stickiness(variants), variants, group_id)
     end
 
-    def resolve_stickiness
-      self.variant_definitions&.map(&:stickiness)&.compact&.first || "default"
+    def resolve_stickiness(variants)
+      variants&.map(&:stickiness)&.compact&.first || "default"
     end
 
     # only check if it is enabled, do not do metrics
     def am_enabled?(context)
-      result =
-        if self.enabled
-          self.strategies.empty? ||
-            self.strategies.any? do |s|
-              strategy_enabled?(s, context) && strategy_constraint_matches?(s, context)
-            end
-        else
-          false
+      evaluate(context)[:is_enabled]
+    end
+
+    def evaluate(context)
+      evaluation_result = {
+        is_enabled: false,
+        strategy: nil
+      }
+      return evaluation_result unless self.enabled
+
+      if self.strategies.empty?
+        evaluation_result[:is_enabled] = true
+      else
+        evaluation_result[:strategy] = self.strategies.find(proc{ nil }) do |s|
+          (strategy_enabled?(s, context) && strategy_constraint_matches?(s, context))
         end
+        evaluation_result[:is_enabled] = true if evaluation_result[:strategy]
+      end
```

Comment:
I feel like there is room for improvement somehow, but I have not yet come up with anything too elegant. Once `FeatureEvaluationResult` issue is addressed, it should be a bit clearer for me.

So I might save that for a possible second round of review 🙈